### PR TITLE
Buffs the SC/FISHER Saboteur Handgun.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
The saboteur gun will now silence pAIs, toggle off radio broadcasting (won't auto-relay nearby speech), disable turrets, chill out secbots a little, and turn off APCs like power outages do. 
The disrupt duration has also been buffed from 10/20 to 15/25 for ranged and point-blank respectively.
Removed a conspicious chat message from an otherwise inconspicious gun.
Brought the code up to date.

## Why It's Good For The Game
The concept is cool, alas it's also undermined by how much of a joke it's right now, and the game has plenty already.
The amount of interactions it has with things is underwhelming, so you could barely consider it a stealth tool. The duration is also quite scarce, I pointed that out in the original PR too.

Basically, I want to make the item cooler.

## Changelog

:cl:
balance: Buffed the duration of the SC/FISHER Saboteur Handgun's disruption effects. It's also stealthier and it won't conspiciously alert living mobs hit by it.
add: Added saboteur interactions with radios, pAIs, turrets, secbots and APCs.
/:cl:

### CODE REVIEW

**Comment by EuclidianGeometer** [comment]
Removing the chat message isn't in the changelog, but it's one of the more important changes here.

**Comment by Ghommie** [comment]
> Removing the chat message isn't in the changelog, but it's one of the more important changes here.

Done.

**Comment by Admiral-Gigaton** [comment]
Can you fix the part where it can't shoot floor lights, but can still melee them?

Also, making it so players with flashlights of any kind, when shot or hit with the gun, have their light turned off. Like a nightmare, but without the breaking of the light. Just shuts it off

**Comment by Eronymun** [comment]
Random suggestion: If you make it turn off (but not lock/discharge) stun batons, I think that would be very silly.

**Comment by rageguy505** [comment]
> Can you fix the part where it can't shoot floor lights, but can still melee them?
> 
> Also, making it so players with flashlights of any kind, when shot or hit with the gun, have their light turned off. Like a nightmare, but without the breaking of the light. Just shuts it off

It already does the second part

**Comment by Admiral-Gigaton** [comment]
> > Can you fix the part where it can't shoot floor lights, but can still melee them?
> > Also, making it so players with flashlights of any kind, when shot or hit with the gun, have their light turned off. Like a nightmare, but without the breaking of the light. Just shuts it off
> 
> It already does the second part

Tested on a recent round (last night) it did not, in fact, seem to disable their lights (plasmaman helmet, flashlights, PDA) while they held them

**Comment by Ghommie** [comment]
> 
> Tested on a recent round (last night) it did not, in fact, seem to disable their lights (plasmaman helmet, flashlights, PDA) while they held them

I'm pretty sure it disables flashlights, but only the ones that can be toggled on/off, so not the fancy subtypes like flares and glowsticks, but I'll take care of stuff like hard hats, plasmaman helmets and PDAs.

**Comment by Admiral-Gigaton** [comment]
> > Tested on a recent round (last night) it did not, in fact, seem to disable their lights (plasmaman helmet, flashlights, PDA) while they held them
> 
> I'm pretty sure it disables flashlights, but only the ones that can be toggled on/off, so not the fancy subtypes like flares and glowsticks, but I'll take care of stuff like hard hats, plasmaman helmets and PDAs.

MODsuit lights too? If possible. Things like eye flashlights wouldn't go on the list tho, as shooting someone's eyes would be a hard task and much more noticeable. 

Mech lights?

**Comment by Admiral-Gigaton** [comment]
should we buff the time for camera EMP? it feels really short, only good for like running across a room

**Comment by Eronymun** [comment]
Haha nice batong go bzzt

File: code/game/objects/items/melee/baton.dm [hunk 1]
```diff
@@ -488,6 +489,14 @@
...
 	qdel(src)
 
+/obj/item/melee/baton/security/proc/on_saboteur(datum/source, disrupt_duration)
+	SIGNAL_HANDLER
+	if(active)
+		toggle_light()
+		active = FALSE
+		update_appearance()
+		return COMSIG_SABOTEUR_SUCCESS
+
```
Comment (Jacquerel) [inline code/game/objects/items/melee/baton.dm:499]:
```suggestion
	if(!active)
		return
	toggle_light()
	active = FALSE
	update_appearance()
	return COMSIG_SABOTEUR_SUCCESS
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]
